### PR TITLE
Fix CANNODE_SUB_MBD typo and broken GPS_UBX_MODE param. Update GPS Submodule. Increase allowed RTK injections.

### PR DESCRIPTION
**Pull request by AlexKlimaj**
Fixes a typo in my last PR for CANNODE_SUB_MBD. Increase the allowed RTK injections to match the uorb queue depth. Update GPS submodule with fixed UBX_MODE enum.

### CODE REVIEW

File: src/drivers/gps/gps.cpp [hunk 1]
```diff
@@ -535,11 +535,12 @@ void GPS::handleInjectDataTopic()
...
 	bool updated = false;
 
-	// Limit maximum number of GPS injections to 6 since usually
+	// Limit maximum number of GPS injections to 12 since usually
 	// GPS injections should consist of 1-4 packets (GPS, Glonass, BeiDou, Galileo).
 	// Looking at 6 packets thus guarantees, that at least a full injection
 	// data set is evaluated.
-	const size_t max_num_injections = 6;
+	// Moving Base reuires a higher rate, so we allow up to 12 packets.
+	const size_t max_num_injections = 12;
```
Comment (dagar) [inline src/drivers/gps/gps.cpp:543]: The queue depth for gps_inject_data is only 8.

Comment (AlexKlimaj) [reply src/drivers/gps/gps.cpp:543]: I'll reduce to 8.

Comment (AlexKlimaj) [reply src/drivers/gps/gps.cpp:543]: Seeing 50Hz injection rate with moving baseline.

**Comment by AlexKlimaj** [comment]
@dagar This needs to be added in the 1.13.2 release.

**Review by dagar — APPROVED** [review]

**Comment by dagar** [comment]
> @dagar This needs to be added in the 1.13.2 release.

Is just the CANNODE_SUB_MBD commit sufficient? Changing the submodule has other implications.

https://github.com/PX4/PX4-Autopilot/pull/20621/commits/a76a7805bc21b0908e0b00593b55197dc1ab9917

**Comment by AlexKlimaj** [comment]
> > @dagar This needs to be added in the 1.13.2 release.
> 
> Is just the CANNODE_SUB_MBD commit sufficient? Changing the submodule has other implications.
> 
> [a76a780](https://github.com/PX4/PX4-Autopilot/commit/a76a7805bc21b0908e0b00593b55197dc1ab9917)

We need the submodule to fix the broken param values.

**Pull request merged**